linked_list.c

```diff
@@ -1,105 +1,197 @@
 #include <assert.h>
 #include "linked_list.h"
 
+
 void ll_init(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
+
+    node->next = NULL;
+    node->prev = NULL;
 }
 
 bool ll_has_next(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
-    return false;
+
+    return node->next != NULL;
 }
 
 bool ll_has_prev(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
-    return false;
+
+    return node->prev != NULL;
 }
 
 struct ll_node *ll_next(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
+
+    if(ll_has_next(node)) {
+        return node->next;
```

Comment:
Since your list is not circular with the head and tail set to `NULL`, it would be safe to simply `return node->next` in all cases.  There's nothing really *wrong* with this approach, but it does introduce unnecessary indirection.